src/ray/core_worker/reference_count.cc

```diff
@@ -56,9 +56,22 @@ void ReferenceCounter::RemoveReferenceRecursive(const ObjectID &object_id) {
       }
     }
     object_id_refs_.erase(object_id);
+    if (on_delete_ != nullptr) {
+      on_delete_(object_id);
+    }
   }
 }
 
+bool ReferenceCounter::HasReference(const ObjectID &object_id) {
+  absl::MutexLock lock(&mutex_);
+  return object_id_refs_.find(object_id) != object_id_refs_.end();
+}
+
```

Comment:
Do we set this dynamically because we initialize the local memory store after the ref counter? If so this should probably append a callback rather than overwrite an existing one if it exists as we will want to perform multiple operations when an object id goes out of scope. That or set one callback into a core worker "HandleObjectOutOfScope" when we construct this.

Reply:
I added a RAY_CHECK that this is only called once, to avoid any correctness concerns.

Reply:
> will want to perform multiple operations

we can add that later.